src/java.base/share/classes/sun/security/ssl/SSLSocketImpl.java

```diff
@@ -1664,8 +1673,9 @@ private void handleException(Exception cause) throws IOException {
             SSLLogger.warning("handling exception", cause);
         }
 
-        // Don't close the Socket in case of timeouts or interrupts.
-        if (cause instanceof InterruptedIOException) {
+        // Don't close the Socket in case of timeouts, interrupts or SocketException.
+        if (cause instanceof InterruptedIOException ||
+                cause instanceof SocketException) {
```

Comment:
Maybe we still need to shutdown the connection with a fatal alter for socket exception, otherwise there might be socket leaks.  Instead, the socket exception could be thrown after the fatal alert.

Reply:
Updated to shutdown the socket in case of a SocketException.